numpy/lib/stride_tricks.py

```diff
@@ -197,6 +197,37 @@ def _broadcast_shape(*args):
     return b.shape
 
 
+def _broadcast_shape_dispatcher(*args):
+    return args
+
+
+@array_function_dispatch(_broadcast_shape_dispatcher, module='numpy')
+def broadcast_shape(*args):
+    """
+    Get Broadcast shape from a list of shape tuples.
+
+    Parameters
+    ----------
+    `*args` : tuples
+        The shapes to be broadcast against each other.
+
+    Returns
+    -------
+    tuple
+        Broadcasted shape.
+
+    Examples
+    --------
+    >>> np.broadcast_shape((1, 2), (3, 1), (3,2))
+    (3, 2)
+
+    >>> np.broadcast_shape((6, 7), (5, 6, 1), (7,), (5, 1, 7))
+    (5, 6, 7)
+    """
+    arrays = list(map(lambda x: np.empty(x), args))
```

Comment:
This would be significantly faster for large shapes if it performed allocations of size 0:
```suggestion
    arrays = list(map(lambda x: np.empty(x, dtype=[]), args)
```
and likely more readable as
```suggestion
    arrays = [np.empty(x, dtype=[]) for x in args]
```

Reply:
Great, thanks so much, apologies, I did see that you suggested dtype=[] in the original issue #17217